### PR TITLE
Enh: Add `isort` and lock in revision on `black`

### LINKED ISSUE #113: Is it worth encouraging isort?
While linting away unused imports is already done in the pre-commit stage, and in CI, I've had a few projects where the team preferred to also use isort to automatically sort imports. 
https://pycqa.github.io/isort/index.html
Are there opinions about this in either direction? Is it worth putting together a brief PR that puts an isort call in the pre-commit? It would behave much like black, and can eventually be added to #105 for CI.

### PR DESCRIPTION
**Pull request by maffettone**
Closes #113 

Updates
=======
- isort
- black revision number
- flake8 source and revision number

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/.pre-commit-config.yaml [hunk 1]
```diff
@@ -2,13 +2,17 @@ default_language_version:
...
 repos:
   - repo: https://github.com/ambv/black
-    rev: stable
+    rev: 22.12.0
     hooks:
       - id: black
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v2.0.0
     hooks:
       - id: flake8
```
Comment (mrakitin) [inline {{ cookiecutter.repo_name }}/.pre-commit-config.yaml:11]: With pre-commit-hooks v4+, this `flake8` option did not work for me. I think it should be updated. That's what I came up with: https://github.com/BNL-ATF/ophyd-basler/blob/main/.pre-commit-config.yaml. Could you please try that config?

**Review by mrakitin — COMMENTED** [review]
Good addition! There may be potential issues with `flake8` (listed below), so I think it's worth fixing here too.

File: {{ cookiecutter.repo_name }}/.pre-commit-config.yaml (hunk 1 above)
Comment (maffettone) [reply {{ cookiecutter.repo_name }}/.pre-commit-config.yaml:11]: Good shout, better to be explicit and not introduce another dependency. Changed below.

Comment (maffettone) [reply {{ cookiecutter.repo_name }}/.pre-commit-config.yaml:11]: I also had a look over the other features in v4.4.0 and nothing else popped out as a "must" include.

**Review by mrakitin — APPROVED** [review]
Looks good! Feel free to self-merge.

**Comment by maffettone** [comment]
Great, updating docs now and will merge when finished

**Review by mrakitin — APPROVED** [review]
❤️ for docs!

**Pull request merged**